src/MappedArrays.jl

```diff
@@ -62,9 +62,10 @@ parenttype{T,N,A,F,Finv}(::Type{MappedArray{T,N,A,F,Finv}}) = A
 
 function testvalue(data)
     if !isempty(data)
-        return first(data)
-    end
-    zero(eltype(data))
+        first(data)
+    else
+        zero(eltype(data))
+    end::eltype(data)
```

Comment:
Zen ponderings: if a container has a looser type than the element type, is it better to return something claimed to be of the container type, the first element type, or the narrowest type that contains all elements? Case: `data = Real[3.2]`.